### PR TITLE
Local Header Bar: Fix various issues and remove markup duplication

### PR DESCRIPTION
**Pull request by shaunandrews**
The PR refactors the "local-header" addressing a handful of issues: #163 #184 #176 #175 #172 #171

--

- Fixes the extra white-space below the Top Bar and the homepage header by adding a color to the body tag.
- Uses Flexbox to avoid having a duplicate header template in every template. (No obvious visual change, but does change some behavior.)
- Ensures the right hover color for links in the Top Bar.
- Adds a transition to the scrim and chevron arrow when toggling the Categories menu.

### CODE REVIEW

**Comment by iandunn** [comment]
There's a simpler solution to removing the white gap: https://github.com/WordPress/wporg-news-2021/issues/163#issuecomment-1009362372

I'd be awesome to remove the need for the duplicate markup, though. If it's not easy to finish that part, it may be better to go with the simple solution for now, and then circle back after the launch, since we're on a tight deadline.

**Comment by shaunandrews** [comment]
> I'd be awesome to remove the need for the duplicate markup, though. If it's not easy to finish that part

I think that part _is_ finished. This could use a review.

File: source/wp-content/themes/wporg-news-2021/sass/components/_local-header.scss [hunk 1]
```diff
@@ -1,130 +1,125 @@
...
-			@include break-xlarge {
-				padding: 0;
-			}
-		}
+		// This overrides a left/right padding using
+		// the `--wp--custom--alignment--edge-spacing` variable.
+		//
+		// The !important is temporary and should  be accomplished
+		// with more specific selectors.
+		// padding: 0 !important;
```
Comment (ryelle) [inline source/wp-content/themes/wporg-news-2021/sass/components/_local-header.scss]: Was this not necessary after all?

File: source/wp-content/themes/wporg-news-2021/sass/components/_local-header.scss [hunk 2]
```diff
@@ -133,104 +128,87 @@
...
-				@include break-mobile {
-					border: none;
-
-					&::before {
-						content: "\0020\00B7\0020";
-					}
+				position: relative;
+
+				// On small viewports the current category is highlighted
+				// using a right-side border.
```
Comment (ryelle) [inline source/wp-content/themes/wporg-news-2021/sass/components/_local-header.scss:205]: is this still in-progress?

**Review by ryelle — COMMENTED** [review]
Overall this is looking great. 💯  for getting rid of the duplicate header.

There are a few places with commented-out code— if you're still working on that, fine, otherwise it should get removed.

I noticed a few issues in the small-screen menu:

- The padding to the right of the Categories label seems too big
- This should be a single column (#172, which is how it looks on trunk to me, so I don't know what that issue is pointing out)

Screen Shot 2022-01-11 at 3 00 38 PM

I also noticed that I can hover and interact with the page behind the overlay when the Categories menu is open, which is not the case on trunk.

File: source/wp-content/themes/wporg-news-2021/sass/components/_local-header.scss (hunk 2 above)
Comment (shaunandrews) [reply source/wp-content/themes/wporg-news-2021/sass/components/_local-header.scss:205]: Oh my. I somehow had changes on my local that weren't pushed to this PR. Sorry about that. Many of these "in-progress" things have been fixed.

**Comment by shaunandrews** [comment]
> There are a few places with commented-out code
> The padding to the right of the Categories label seems too big
> This should be a single column (#172, which is how it looks on trunk to me, so I don't know what that issue is pointing out)

I must have forgotten to push some changes. This should all be fixed.

> I also noticed that I can hover and interact with the page behind the overlay when the Categories menu is open, which is not the case on trunk.

I just pushed a fix for this.

File: source/wp-content/themes/wporg-news-2021/sass/components/_local-header.scss [hunk 3]
```diff
@@ -133,102 +130,116 @@
...
-						position: fixed;
-						top: 0;
-						left: 0;
-						width: 100%;
-						height: 100vh;
-						background-color: rgba(0, 0, 0, 0.5);
-						z-index: -1;
-						animation-duration: 400ms;
-						animation-name: overlay-fade-in;
+						opacity: 1;
```
Comment (ryelle) [inline source/wp-content/themes/wporg-news-2021/sass/components/_local-header.scss:136]: I think display would work better here - with `opacity` it's always on screen, so it's overlaying the whole content even when the menu is closed.

File: source/wp-content/themes/wporg-news-2021/sass/components/_site-header.scss [hunk 4]
```diff
@@ -1,33 +1,18 @@
...
+			position: sticky;
+			top: 0;
 
-		/*
-		 * For some reason this element is taller in the offset header than it is in the regular header. That adds
-		 * extra spacing after the header that we don't want.
-		 */
-		display: none;
+			body.admin-bar & {
+				padding-top: var(--wpadmin-bar--height);
```
Comment (ryelle) [inline source/wp-content/themes/wporg-news-2021/sass/components/_site-header.scss]: There's actually a core-provided value for this now, so you can use `--wp-admin--admin-bar--height`
```suggestion
				padding-top: var(--wp-admin--admin-bar--height, 0);
```

**Review by ryelle — COMMENTED** [review]
Looking much better with the rest of your changes 😁  all the things I mentioned before are fixed.

Are the white dots here just part of the brush svg? it seems odd since it kind of lines up with the text, but I don't know if I'd go as far as saying it's a bug.
Screen Shot 2022-01-12 at 1 29 29 PM

The hover/focus colors seem off on the non-blue schemes, I don't know if those were designed.

1. When the text is black, the hover is still light:
    Screen Shot 2022-01-12 at 1 33 08 PM
2. On the black bar, there is no visible hover on "News"
    Screen Shot 2022-01-12 at 1 34 09 PM
3. The little ▪️ is not visible on the black menu (or in the bar ^)
    Screen Shot 2022-01-12 at 1 34 17 PM

**Comment by ryelle** [comment]
Oh also, can you run `yarn workspace wporg-news-2021-theme lint:css --fix` to fix up the linter issues? thanks!

File: source/wp-content/themes/wporg-news-2021/sass/components/_local-header.scss (hunk 3 above)
Comment (shaunandrews) [reply source/wp-content/themes/wporg-news-2021/sass/components/_local-header.scss:136]: We're able to transition the value of `opacity`, but we can't do that if also has `display: none;`. Are there implication to having it there (but transparent) all the time?

Comment (ryelle) [reply source/wp-content/themes/wporg-news-2021/sass/components/_local-header.scss:136]: Yeah, clicking anywhere on the page opens the menu :)

Comment (shaunandrews) [reply source/wp-content/themes/wporg-news-2021/sass/components/_local-header.scss:136]: lol, really? How did I miss that... I'll look into it.

Comment (shaunandrews) [reply source/wp-content/themes/wporg-news-2021/sass/components/_local-header.scss:136]: My bad. Fixed now. `pointer-events` only work when the menu is open now.

Comment (ryelle) [reply source/wp-content/themes/wporg-news-2021/sass/components/_local-header.scss:136]: 👍🏻  yep, working for me now

**Comment by shaunandrews** [comment]
> Are the white dots here just part of the brush svg? it seems odd since it kind of lines up with the text, but I don't know if I'd go as far as saying it's a bug.

Its part of the SVG mask. I've re-exported the mask with the hope of better reflecting the design.

> The hover/focus colors seem off on the non-blue schemes, I don't know if those were designed.

These should all be resolved now. There are some basics in the Figma (https://www.figma.com/file/O9I8Jl7kaUMTX39LlRGbGO/News-Page?node-id=4659%3A8224) but I'm not sure they're 100% correct.

**Review by ryelle — APPROVED** [review]
Looks good now — I just pushed a quick change to remove the core-provided margin for the admin bar, which fixes this issue:

Screen Shot 2022-01-12 at 4 41 03 PM

**Comment by ryelle** [comment]
> Its part of the SVG mask. I've re-exported the mask with the hope of better reflecting the design.

Definitely better now, thanks!

**Comment by shaunandrews** [comment]
> remove the core-provided margin for the admin bar

Perfect; I knew there was a way to do it! Thanks.

**Pull request merged**